SPEC.md

```diff
@@ -406,3 +406,38 @@ User Label fields:
 * colour
 * user
 * team - either team or user must be set
+
+# Search
+
+Should separate search logic now to make elasticsearch integration in future easier?
+
+Search entries have:
+* the conv id
+* a list of participants
+* the search vector
+* creator id
+* participant ids
+* the last action the refer to (nullable)
+* might need language too in time
+
+When a participant is removed a search entry is created (or used, hence the last action field) and the
+participant is added to it. Thus people can still search for conversations they're removed from.
+
+Main search entry for a conversation holds a tsv:
+* when a message is added new tsv is appended to existing one
+* when message is edited or subject changed etc., the whole tsv is rebuilt
+* participant email address should be included in the tsv with a higher weight, also domains `@example.com`
+  to allow such searches
```

Comment:
weights can also be used to search on a particular thing, eg.:
* A: subject
* B: email addresses of participants
* C: messages
* D: files

This if someone searches `includes:foobar@example.com` the search can be constrained to weight `B`.